pkg/trace/obfuscate/sql.go

```diff
@@ -342,3 +342,23 @@ func (o *Obfuscator) obfuscateSQL(span *pb.Span) {
 	}
 	traceutil.SetMeta(span, sqlQueryTag, oq.Query)
 }
+
+// ObfuscateSQLStringSafe obfuscates the sql string, returning either the successfully obfuscated string or a string
+// containing an error message explaining why the obfuscation failed
+func (o *Obfuscator) ObfuscateSQLStringSafe(sqlString string) string {
+	result, err := o.ObfuscateSQLString(sqlString)
+	if err != nil {
+		log.Debugf("failed to obfuscate sql string '%s': %s", sqlString, err.Error())
+		return "datadog-agent failed to obfuscate sql string. enable agent debug logs for more info."
```

Comment:
Is there any point in putting `err.Error()` here?

Reply:
No, the point is that this can only return a string. If there's an error then we put the error message in the string itself. That way when viewing the query that failed to obfuscate in our product it's clear that there's a problem and it's not just empty or missing. 
https://github.com/DataDog/datadog-agent/pull/5692/commits/65e0e51eff80d093aa80995bdb7c87e43007b628